source/common/event/dispatcher_impl.cc

```diff
@@ -119,9 +119,9 @@ DispatcherImpl::createSslListener(Network::ConnectionHandler& conn_handler,
 }
 
 TimerPtr DispatcherImpl::createTimer(TimerCb cb) {
-  // TODO(htuch): Figure out why we can't set this assertion, see
-  // https://github.com/lyft/envoy/issues/1294.
-  // ASSERT(isThreadSafe());
+  // We don't ASSERT that createTimer() is called on the run thread because timers are used for
```

Comment:
A few things:
1) post() and deferredDelete() are actually implemented in terms of timers, and those are setup before we run on another thread for obvious reasons: https://github.com/lyft/envoy/blob/master/source/common/event/dispatcher_impl.cc#L33. Initially, I had a change where those two timer creates would not fire the assert.
2) However, there is at least one other place where we use a timer for cross thread sync without ever doing post: https://github.com/lyft/envoy/blob/master/source/common/filesystem/filesystem_impl.cc#L187. I think I could fix <- but I'm not sure it's really worth it right now since this code is technically fine and all thread safe.

LMK what you think. I can add more comments and TODO() also if you want.